viadot/sources/genesys.py

```diff
@@ -187,17 +176,21 @@ async def generate_post():
                                     f"Generated report export --- \n {payload}."
                                 )
                                 semaphore.release()
-                    cnt += 1
-                else:
-                    await asyncio.sleep(3)
-                    cnt = 0
+                        elif method == "GET":
+                            async with session.get(
+                                url,
+                                headers=self.authorization_token,
+                                params=params,
+                            ) as resp:
+                                new_report = await resp.read()
+                                semaphore.release()
+                await asyncio.sleep(0.5)
```

Comment:
Does it need to be documented?
Maybe better to bring timeout-setting into arguments?

Reply:
What do you mean by documented?
The timeout parameter is passed at the task level, where do you suggest to put into the code?

Reply:
it is good to give it as an argument to a method and describing as other arguments.

Reply:
I mean `sleep_timeout = 0.5` as an argument.